Add new components:

AnasTomeh opened the pull request:
1- Button Component
2- Topic List Component
3- Welcome Bar Component

2- Modifying Existing Components
3- Enhance layout

AnasTomeh committed:
Add new components:
1- Button Component
2- Topic List Component
3- Welcome Bar Component

2- Modifying Existing Components
3- Enhance layout

Hartha-aloufi commented on src/App.js:
```diff
@@ -61,40 +63,30 @@ function App() {
         <>
-            
             <BrowserRouter>
+            <Header
+                toggleDarkMode={toggleDarkMode}
+                isDarkModeTheme={isDarkModeTheme}
+                onShowModal={handleDisplayingModal}
+            />
+            <WelcomeBar/>
```
Would see this in the `Layout` component.

Layout component has this skelatone: 

```
const Layout = () => {
return (
<div className="layout">
    <Header />

    <main className="content">
      {Outlet}
    </main>

   <Footer />
</div>
)

}
```

Hartha-aloufi commented on src/App.js:
```diff
@@ -61,40 +63,30 @@ function App() {
-                        onShowModal={handleDisplayingModal}
-                        />
-                    }
+                    <Route exact path="Anas-Tomeh-Project1-React/" element={<HomePage/>}
```
if we considered `Anas-Tomeh-Project1-React` as the domain name, then this is wrong. the path should be the thing that is after the domain name. 

ex: Github  domain name is : `https://github.com/`.
index page path should be: `path="/"`
If github has a detailed page, then the path should be: `path="/details/${topicId}"`

Hartha-aloufi commented on src/App.js:
```diff
@@ -61,40 +63,30 @@ function App() {
+            <Footer/>
             <Modal isOpen={modalShow} title={'My Favourite Topics'}>
                 <FavouriteTopicCard favouriteTopics={favouriteTopics}/>
             </Modal>
```
Modal thing should not be here, move it to a component file

Hartha-aloufi commented on src/components/Cards/TopicCard/TopicCard.js:
```diff
@@ -0,0 +1,34 @@
+                           <h2 className="topic-category">{topic.category}</h2>
+                           <p className="topic-name">{topic.topic}</p>
+                           <div className="rating-stars">
+                               <p>★★★★★</p>
```
why it is not dynamic

AnasTomeh replied:
I need to merge the main branch to this branch so I can use StarIcon comp then i will handle it

Hartha-aloufi commented on src/components/Cards/TopicCard/TopicCard.js:
```diff
@@ -0,0 +1,34 @@
+const TopicCard = ({topic}) => {
+
+    return (
+       <>
```
why you need this fragment?

AnasTomeh replied:
because it have to child:
1- {isLoading && <LoadingIndicator message={'Topic is Loading ...'}/>}
           2-  <div>
                {topic &&

Hartha-aloufi commented on src/components/Header/Header.js:
```diff
@@ -3,7 +3,12 @@ import './Header.css'
 import {useEffect, useState} from "react";
 import {Link} from "react-router-dom";
-const Header = ({toggleDarkMode, isDarkModeTheme, onShowModal}) => {
+import WelcomeBar from "../WelcomeBar/WelcomeBar";
```
remove this import if you don't use it

Hartha-aloufi commented on src/components/Header/Header.js:
```diff
@@ -3,7 +3,12 @@ import './Header.css'
+
+    const {toggleDarkMode, isDarkModeTheme, onShowModal} = props
 
     const [screenWidth, setScreenWidth] = useState(window.innerWidth);
```
This one should not be saved in a state, since it will not notify you when the screen size changed. 

take a look at this: https://usehooks-ts.com/react-hook/use-media-query

AnasTomeh committed:
Create & use useApi custom hook

AnasTomeh committed:
1- Create ThemeContext and enhance dark mode handling
2- Create UseApi and use it for fetching

AnasTomeh committed:
1- resolve page path
2- move modal from app.js
3- create and use useMediaQuery to handle mobile design
3- create RatingBar Comp
